Adapt and add info on constraints in docs and docstrings (#251)
rl-institut/oemof-B3: docs/model_structure.rst

    :backlinks: top

In oemof-B3, data appears in different formats in each processing step. Here, we give a short
overview. In general, it should be noted that energy carriers are not the same as busses.
For example, electricity can be generated from the energy carrier biomass without biomass being added as a bus.

> The busses used in addition to the electricity bus in oemof-B3 are given in :file:`bus_attrs_update.yml`.

Reviewer: This information seems misplaced here, at least in my view.

Raw data
--------

> Raw data from external source comes in different formats. It is provided in :file:`raw/`.

Reviewer: ```suggestion
Raw data from external source comes in different formats. It is not part of the model on GitHub, but has to be downloaded separately to the directory :file:`raw/`.
```

   :delim: ;
   :file: ../oemof_b3/schema/timeseries.csv

> Within oemof-B3 resources are located in :file:`results/_resources`.

Reviewer: ```suggestion
Within oemof-B3, resources are located in :file:`results/_resources`.
```

A few more conventions are important to know. Missing data is left empty. If a value applies to all
regions, this is indicated by :attr:`ALL`. If it applies to the sum of regions, by :attr:`TOTAL`.
There is no unit transformation within the model, i.e. the user needs to ensure the consistency of units.

Further information about specific parameters should be known.

> Components can receive keywords for the electricity gas relation constraint in variable :attr:`output_parameters`.

Reviewer: ```suggestion
Components can receive keywords for the electricity-gas-relation-constraint via the attribute :attr:`output_parameters`.
```

Keywords of components powered by gas start with :attr:`config.settings.optimize.gas_key` and such powered
with electricity with :attr:`config.settings.optimize.el_key` followed by :attr:`carrier` and :attr:`region` (example: :attr:`{"electricity-heat_decentral-B": 1}`).

> Do not provide :attr:`output_parameters` or leave their :attr:`var_value` empty to neglect a component in the constraint.

Reviewer: This is hard to understand.

Preprocessed datapackages
-------------------------

The resources are then again preprocessed together with the scenario information to generate
scenario-specific datapackages. A preprocessed datapackage represents an instance of an energy system scenario.
It is a collection of .csv-files, one file for all busses and one for each

> component, stored in :file:`results/<scneario_key>/preprocessed/data/elements` (scalars data) and

Reviewer: ```suggestion
component, stored in :file:`results/<scenario>/preprocessed/data/elements` (scalar data) and
```

> :file:`results/<scneario_key>/preprocessed/data/sequences` (time series for e.g. renewable feedin or demand profiles),

Reviewer: ```suggestion
:file:`results/<scenario>/preprocessed/data/sequences` (time series for e.g. renewable feed-in or demand profiles),
```

regions, this is indicated by :attr:`ALL`. If it applies to the sum of regions, by :attr:`TOTAL`.
There is no unit transformation within the model, i.e. the user needs to ensure the consistency of units.

> Further information about specific parameters should be known.

Reviewer 2: ```suggestion
Further information about specific parameters should be considered.
```

> More generally, there are specific variables which depend on the type of the component. Components and
> their properties are defined in
> `oemoflex <https://github.com/rl-institut/oemoflex/tree/dev/oemoflex/model>`_.

Reviewer: ```suggestion
More generally, there are specific variables which depend on the type of the component. Components and
their attributes are defined in
`oemoflex <https://github.com/rl-institut/oemoflex/tree/dev/oemoflex/model/component_attrs.yml>`_.
```

Components and properties can also be added or updated in oemof-B3 using the files in :file:`oemof_b3/model/`.
You can have a look at the structure of a datapackage using our example:

> `datapackage of "example_base" <https://github.com/rl-institut/oemof-B3/tree/dev/examples/example_base/preprocessed>`_.

Reviewer: This should be at the top of this section.
